{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "aa9cefd4",
   "metadata": {},
   "source": [
    "# Library"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7dec6ed4",
   "metadata": {},
   "source": [
    "## Python packages "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "dca3bc93",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/tanya/anaconda3/lib/python3.11/site-packages/pandas/core/arrays/masked.py:60: UserWarning: Pandas requires version '1.3.6' or newer of 'bottleneck' (version '1.3.5' currently installed).\n",
      "  from pandas.core import (\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import regex as re\n",
    "import rpy2\n",
    "import os\n",
    "import functools as fct\n",
    "from collections import Counter\n",
    "import pickle "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "04ddeb7d",
   "metadata": {},
   "source": [
    "## R packages "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "86743d88",
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "3a5a43e6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "Attaching package: ‘dplyr’\n",
       "\n",
       "The following objects are masked from ‘package:stats’:\n",
       "\n",
       "    filter, lag\n",
       "\n",
       "The following objects are masked from ‘package:base’:\n",
       "\n",
       "    intersect, setdiff, setequal, union\n",
       "\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "library(rlang)\n",
    "library(ggplot2)\n",
    "library(dplyr)\n",
    "library(scales)\n",
    "library(reshape2)\n",
    "library(cowplot)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "815cc4df",
   "metadata": {},
   "source": [
    "## Set paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bf6cb607",
   "metadata": {},
   "outputs": [],
   "source": [
    "! mkdir plots tables temp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "a141a813",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Set paths \n",
    "\n",
    "path_to_data = '/Users/tanya/Documents/OTT/Conflicts_ClinVar/data' \n",
    "path_to_vcfs = '/Users/tanya/Documents/OTT/Conflicts_ClinVar/data/vcfs'   # path to initial ClinVar dfs\n",
    "path_to_gatk = '/Users/tanya/Documents/tools/gatk-4.5.0.0/gatk'\n",
    "path_to_tables = '/Users/tanya/Documents/OTT/Conflicts_ClinVar/tables'\n",
    "path_to_plots = '/Users/tanya/Documents/OTT/Conflicts_ClinVar/plots'\n",
    "path_to_temp = '/Users/tanya/Documents/OTT/Conflicts_ClinVar/temp'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b22c7d36",
   "metadata": {},
   "source": [
    "# Download data\n",
    "\n",
    "Source: https://ftp.ncbi.nlm.nih.gov/pub/clinvar/  \n",
    "Assemby: GRCh38  \n",
    "Period: every 6 months between 2018 and 2024"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "8ea8cd63",
   "metadata": {},
   "outputs": [],
   "source": [
    "# List of all available VCFs\n",
    "\n",
    "vcfs_all = ['clinvar_20140401', 'clinvar_20140929', 'clinvar_20150330', 'clinvar_20150929', \n",
    "        'clinvar_20160405', 'clinvar_20161003', 'clinvar_20170516', 'clinvar_20171002',\n",
    "        'clinvar_20180401', 'clinvar_20180930', 'clinvar_20190403', 'clinvar_20191001',\n",
    "        'clinvar_20200407', 'clinvar_20200927', 'clinvar_20210328', 'clinvar_20210929',\n",
    "        'clinvar_20220403', 'clinvar_20221001', 'clinvar_20230404',  'clinvar_20230930',\n",
    "        'clinvar_20240407']\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "eed725ac",
   "metadata": {},
   "outputs": [],
   "source": [
    "# List of VCFs with fields \"Review Status\" and \"Variation IDs\"\n",
    "\n",
    "vcfs = [ 'clinvar_20180401', 'clinvar_20180930', 'clinvar_20190403', 'clinvar_20191001',\n",
    "        'clinvar_20200407', 'clinvar_20200927', 'clinvar_20210328', 'clinvar_20210929',\n",
    "        'clinvar_20220403', 'clinvar_20221001', 'clinvar_20230404',  'clinvar_20230930',\n",
    "        'clinvar_20240407'] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "aa41fd87",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Using GATK jar /Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar VariantsToTable -V /Users/tanya/Documents/OTT/Conflicts_ClinVar/data/vcfs/clinvar_20180401.vcf -F CHROM -F POS -F ID -F REF -F ALT -F QUAL -F FILTER -F GENEINFO -F CLNVC -F CLNSIG -F CLNREVSTAT -F CLNSIGCONF -F CLNDN -F CLNDISDB -O /Users/tanya/Documents/OTT/Conflicts_ClinVar/tables/clinvar_20180401.txt\n",
      "20:37:06.202 INFO  NativeLibraryLoader - Loading libgkl_compression.dylib from jar:file:/Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.dylib\n",
      "20:37:06.236 WARN  NativeLibraryLoader - Unable to load libgkl_compression.dylib from native/libgkl_compression.dylib (/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression17454506857951671673.dylib: dlopen(/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression17454506857951671673.dylib, 0x0001): tried: '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression17454506857951671673.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression17454506857951671673.dylib' (no such file), '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression17454506857951671673.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')))\n",
      "20:37:06.238 INFO  NativeLibraryLoader - Loading libgkl_compression.dylib from jar:file:/Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.dylib\n",
      "20:37:06.242 WARN  NativeLibraryLoader - Unable to load libgkl_compression.dylib from native/libgkl_compression.dylib (/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression18164534222872205612.dylib: dlopen(/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression18164534222872205612.dylib, 0x0001): tried: '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression18164534222872205612.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression18164534222872205612.dylib' (no such file), '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression18164534222872205612.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')))\n",
      "20:37:06.314 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:37:06.317 INFO  VariantsToTable - The Genome Analysis Toolkit (GATK) v4.5.0.0\n",
      "20:37:06.317 INFO  VariantsToTable - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "20:37:06.317 INFO  VariantsToTable - Executing as tanya@Tatianas-MacBook-Air.local on Mac OS X v14.1.2 aarch64\n",
      "20:37:06.317 INFO  VariantsToTable - Java runtime: Java HotSpot(TM) 64-Bit Server VM v21+35-LTS-2513\n",
      "20:37:06.317 INFO  VariantsToTable - Start Date/Time: April 9, 2024, 8:37:06 PM MSK\n",
      "20:37:06.317 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:37:06.317 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:37:06.318 INFO  VariantsToTable - HTSJDK Version: 4.1.0\n",
      "20:37:06.318 INFO  VariantsToTable - Picard Version: 3.1.1\n",
      "20:37:06.318 INFO  VariantsToTable - Built for Spark Version: 3.5.0\n",
      "20:37:06.318 INFO  VariantsToTable - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "20:37:06.318 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "20:37:06.318 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "20:37:06.318 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "20:37:06.318 INFO  VariantsToTable - Deflater: JdkDeflater\n",
      "20:37:06.318 INFO  VariantsToTable - Inflater: JdkInflater\n",
      "20:37:06.319 INFO  VariantsToTable - GCS max retries/reopens: 20\n",
      "20:37:06.319 INFO  VariantsToTable - Requester pays: disabled\n",
      "20:37:06.319 INFO  VariantsToTable - Initializing engine\n",
      "20:37:06.410 INFO  FeatureManager - Using codec VCFCodec to read file file:///Users/tanya/Documents/OTT/Conflicts_ClinVar/data/vcfs/clinvar_20180401.vcf\n",
      "20:37:06.428 INFO  VariantsToTable - Done initializing engine\n",
      "20:37:06.430 WARN  VariantsToTable - Allele-specific fields will only be split if splitting multi-allelic variants is specified (`--split-multi-allelic` or `-SMA`\n",
      "20:37:06.430 INFO  ProgressMeter - Starting traversal\n",
      "20:37:06.431 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "20:37:10.012 INFO  ProgressMeter -              MT:7041              0.1                354334        5941878.1\n",
      "20:37:10.012 INFO  ProgressMeter - Traversal complete. Processed 354334 total variants in 0.1 minutes.\n",
      "20:37:10.012 INFO  VariantsToTable - Shutting down engine\n",
      "[April 9, 2024, 8:37:10 PM MSK] org.broadinstitute.hellbender.tools.walkers.variantutils.VariantsToTable done. Elapsed time: 0.06 minutes.\n",
      "Runtime.totalMemory()=219152384\n",
      "/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/ipykernel_10295/4120375074.py:11: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  cv_dict[vcf] = pd.read_csv(f'{path_to_tables}/{vcf}.txt', delimiter='\\t', comment='#', skip_blank_lines=True)\n",
      "Using GATK jar /Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar VariantsToTable -V /Users/tanya/Documents/OTT/Conflicts_ClinVar/data/vcfs/clinvar_20180930.vcf -F CHROM -F POS -F ID -F REF -F ALT -F QUAL -F FILTER -F GENEINFO -F CLNVC -F CLNSIG -F CLNREVSTAT -F CLNSIGCONF -F CLNDN -F CLNDISDB -O /Users/tanya/Documents/OTT/Conflicts_ClinVar/tables/clinvar_20180930.txt\n",
      "20:37:12.029 INFO  NativeLibraryLoader - Loading libgkl_compression.dylib from jar:file:/Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.dylib\n",
      "20:37:12.062 WARN  NativeLibraryLoader - Unable to load libgkl_compression.dylib from native/libgkl_compression.dylib (/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression7192331710299159843.dylib: dlopen(/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression7192331710299159843.dylib, 0x0001): tried: '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression7192331710299159843.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression7192331710299159843.dylib' (no such file), '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression7192331710299159843.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')))\n",
      "20:37:12.064 INFO  NativeLibraryLoader - Loading libgkl_compression.dylib from jar:file:/Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.dylib\n",
      "20:37:12.067 WARN  NativeLibraryLoader - Unable to load libgkl_compression.dylib from native/libgkl_compression.dylib (/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression7798865867031977562.dylib: dlopen(/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression7798865867031977562.dylib, 0x0001): tried: '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression7798865867031977562.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression7798865867031977562.dylib' (no such file), '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression7798865867031977562.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')))\n",
      "20:37:12.134 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:37:12.136 INFO  VariantsToTable - The Genome Analysis Toolkit (GATK) v4.5.0.0\n",
      "20:37:12.136 INFO  VariantsToTable - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "20:37:12.136 INFO  VariantsToTable - Executing as tanya@Tatianas-MacBook-Air.local on Mac OS X v14.1.2 aarch64\n",
      "20:37:12.136 INFO  VariantsToTable - Java runtime: Java HotSpot(TM) 64-Bit Server VM v21+35-LTS-2513\n",
      "20:37:12.137 INFO  VariantsToTable - Start Date/Time: April 9, 2024, 8:37:12 PM MSK\n",
      "20:37:12.137 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:37:12.137 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:37:12.137 INFO  VariantsToTable - HTSJDK Version: 4.1.0\n",
      "20:37:12.137 INFO  VariantsToTable - Picard Version: 3.1.1\n",
      "20:37:12.137 INFO  VariantsToTable - Built for Spark Version: 3.5.0\n",
      "20:37:12.138 INFO  VariantsToTable - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "20:37:12.138 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "20:37:12.138 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "20:37:12.138 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "20:37:12.138 INFO  VariantsToTable - Deflater: JdkDeflater\n",
      "20:37:12.138 INFO  VariantsToTable - Inflater: JdkInflater\n",
      "20:37:12.138 INFO  VariantsToTable - GCS max retries/reopens: 20\n",
      "20:37:12.138 INFO  VariantsToTable - Requester pays: disabled\n",
      "20:37:12.139 INFO  VariantsToTable - Initializing engine\n",
      "20:37:12.213 INFO  FeatureManager - Using codec VCFCodec to read file file:///Users/tanya/Documents/OTT/Conflicts_ClinVar/data/vcfs/clinvar_20180930.vcf\n",
      "20:37:12.224 INFO  VariantsToTable - Done initializing engine\n",
      "20:37:12.225 WARN  VariantsToTable - Allele-specific fields will only be split if splitting multi-allelic variants is specified (`--split-multi-allelic` or `-SMA`\n",
      "20:37:12.226 INFO  ProgressMeter - Starting traversal\n",
      "20:37:12.226 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "20:37:16.283 INFO  ProgressMeter -          X:154773245              0.1                408919        6053575.1\n",
      "20:37:16.283 INFO  ProgressMeter - Traversal complete. Processed 408919 total variants in 0.1 minutes.\n",
      "20:37:16.283 INFO  VariantsToTable - Shutting down engine\n",
      "[April 9, 2024, 8:37:16 PM MSK] org.broadinstitute.hellbender.tools.walkers.variantutils.VariantsToTable done. Elapsed time: 0.07 minutes.\n",
      "Runtime.totalMemory()=219152384\n",
      "/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/ipykernel_10295/4120375074.py:11: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  cv_dict[vcf] = pd.read_csv(f'{path_to_tables}/{vcf}.txt', delimiter='\\t', comment='#', skip_blank_lines=True)\n",
      "Using GATK jar /Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar VariantsToTable -V /Users/tanya/Documents/OTT/Conflicts_ClinVar/data/vcfs/clinvar_20190403.vcf -F CHROM -F POS -F ID -F REF -F ALT -F QUAL -F FILTER -F GENEINFO -F CLNVC -F CLNSIG -F CLNREVSTAT -F CLNSIGCONF -F CLNDN -F CLNDISDB -O /Users/tanya/Documents/OTT/Conflicts_ClinVar/tables/clinvar_20190403.txt\n",
      "20:37:18.435 INFO  NativeLibraryLoader - Loading libgkl_compression.dylib from jar:file:/Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.dylib\n",
      "20:37:18.468 WARN  NativeLibraryLoader - Unable to load libgkl_compression.dylib from native/libgkl_compression.dylib (/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression8596791795142973936.dylib: dlopen(/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression8596791795142973936.dylib, 0x0001): tried: '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression8596791795142973936.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression8596791795142973936.dylib' (no such file), '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression8596791795142973936.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')))\n",
      "20:37:18.471 INFO  NativeLibraryLoader - Loading libgkl_compression.dylib from jar:file:/Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.dylib\n",
      "20:37:18.474 WARN  NativeLibraryLoader - Unable to load libgkl_compression.dylib from native/libgkl_compression.dylib (/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression11134220974531684864.dylib: dlopen(/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression11134220974531684864.dylib, 0x0001): tried: '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression11134220974531684864.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression11134220974531684864.dylib' (no such file), '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression11134220974531684864.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')))\n",
      "20:37:18.548 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:37:18.550 INFO  VariantsToTable - The Genome Analysis Toolkit (GATK) v4.5.0.0\n",
      "20:37:18.550 INFO  VariantsToTable - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "20:37:18.550 INFO  VariantsToTable - Executing as tanya@Tatianas-MacBook-Air.local on Mac OS X v14.1.2 aarch64\n",
      "20:37:18.550 INFO  VariantsToTable - Java runtime: Java HotSpot(TM) 64-Bit Server VM v21+35-LTS-2513\n",
      "20:37:18.550 INFO  VariantsToTable - Start Date/Time: April 9, 2024, 8:37:18 PM MSK\n",
      "20:37:18.550 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:37:18.550 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:37:18.551 INFO  VariantsToTable - HTSJDK Version: 4.1.0\n",
      "20:37:18.551 INFO  VariantsToTable - Picard Version: 3.1.1\n",
      "20:37:18.551 INFO  VariantsToTable - Built for Spark Version: 3.5.0\n",
      "20:37:18.551 INFO  VariantsToTable - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "20:37:18.551 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "20:37:18.551 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "20:37:18.552 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "20:37:18.552 INFO  VariantsToTable - Deflater: JdkDeflater\n",
      "20:37:18.552 INFO  VariantsToTable - Inflater: JdkInflater\n",
      "20:37:18.552 INFO  VariantsToTable - GCS max retries/reopens: 20\n",
      "20:37:18.552 INFO  VariantsToTable - Requester pays: disabled\n",
      "20:37:18.552 INFO  VariantsToTable - Initializing engine\n",
      "20:37:18.630 INFO  FeatureManager - Using codec VCFCodec to read file file:///Users/tanya/Documents/OTT/Conflicts_ClinVar/data/vcfs/clinvar_20190403.vcf\n",
      "20:37:18.641 INFO  VariantsToTable - Done initializing engine\n",
      "20:37:18.642 WARN  VariantsToTable - Allele-specific fields will only be split if splitting multi-allelic variants is specified (`--split-multi-allelic` or `-SMA`\n",
      "20:37:18.642 INFO  ProgressMeter - Starting traversal\n",
      "20:37:18.642 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "20:37:23.178 INFO  ProgressMeter -               MT:869              0.1                445565        5897617.5\n",
      "20:37:23.178 INFO  ProgressMeter - Traversal complete. Processed 445565 total variants in 0.1 minutes.\n",
      "20:37:23.178 INFO  VariantsToTable - Shutting down engine\n",
      "[April 9, 2024, 8:37:23 PM MSK] org.broadinstitute.hellbender.tools.walkers.variantutils.VariantsToTable done. Elapsed time: 0.08 minutes.\n",
      "Runtime.totalMemory()=218103808\n",
      "/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/ipykernel_10295/4120375074.py:11: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  cv_dict[vcf] = pd.read_csv(f'{path_to_tables}/{vcf}.txt', delimiter='\\t', comment='#', skip_blank_lines=True)\n",
      "Using GATK jar /Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar VariantsToTable -V /Users/tanya/Documents/OTT/Conflicts_ClinVar/data/vcfs/clinvar_20191001.vcf -F CHROM -F POS -F ID -F REF -F ALT -F QUAL -F FILTER -F GENEINFO -F CLNVC -F CLNSIG -F CLNREVSTAT -F CLNSIGCONF -F CLNDN -F CLNDISDB -O /Users/tanya/Documents/OTT/Conflicts_ClinVar/tables/clinvar_20191001.txt\n",
      "20:37:25.360 INFO  NativeLibraryLoader - Loading libgkl_compression.dylib from jar:file:/Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.dylib\n",
      "20:37:25.393 WARN  NativeLibraryLoader - Unable to load libgkl_compression.dylib from native/libgkl_compression.dylib (/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression6261502733298327643.dylib: dlopen(/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression6261502733298327643.dylib, 0x0001): tried: '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression6261502733298327643.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression6261502733298327643.dylib' (no such file), '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression6261502733298327643.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')))\n",
      "20:37:25.396 INFO  NativeLibraryLoader - Loading libgkl_compression.dylib from jar:file:/Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.dylib\n",
      "20:37:25.399 WARN  NativeLibraryLoader - Unable to load libgkl_compression.dylib from native/libgkl_compression.dylib (/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression8871458840619117557.dylib: dlopen(/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression8871458840619117557.dylib, 0x0001): tried: '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression8871458840619117557.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression8871458840619117557.dylib' (no such file), '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression8871458840619117557.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')))\n",
      "20:37:25.463 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:37:25.465 INFO  VariantsToTable - The Genome Analysis Toolkit (GATK) v4.5.0.0\n",
      "20:37:25.465 INFO  VariantsToTable - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "20:37:25.465 INFO  VariantsToTable - Executing as tanya@Tatianas-MacBook-Air.local on Mac OS X v14.1.2 aarch64\n",
      "20:37:25.465 INFO  VariantsToTable - Java runtime: Java HotSpot(TM) 64-Bit Server VM v21+35-LTS-2513\n",
      "20:37:25.465 INFO  VariantsToTable - Start Date/Time: April 9, 2024, 8:37:25 PM MSK\n",
      "20:37:25.465 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:37:25.465 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:37:25.466 INFO  VariantsToTable - HTSJDK Version: 4.1.0\n",
      "20:37:25.466 INFO  VariantsToTable - Picard Version: 3.1.1\n",
      "20:37:25.466 INFO  VariantsToTable - Built for Spark Version: 3.5.0\n",
      "20:37:25.466 INFO  VariantsToTable - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "20:37:25.466 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "20:37:25.466 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "20:37:25.467 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "20:37:25.467 INFO  VariantsToTable - Deflater: JdkDeflater\n",
      "20:37:25.467 INFO  VariantsToTable - Inflater: JdkInflater\n",
      "20:37:25.467 INFO  VariantsToTable - GCS max retries/reopens: 20\n",
      "20:37:25.467 INFO  VariantsToTable - Requester pays: disabled\n",
      "20:37:25.467 INFO  VariantsToTable - Initializing engine\n",
      "20:37:25.541 INFO  FeatureManager - Using codec VCFCodec to read file file:///Users/tanya/Documents/OTT/Conflicts_ClinVar/data/vcfs/clinvar_20191001.vcf\n",
      "20:37:25.552 INFO  VariantsToTable - Done initializing engine\n",
      "20:37:25.552 WARN  VariantsToTable - Allele-specific fields will only be split if splitting multi-allelic variants is specified (`--split-multi-allelic` or `-SMA`\n",
      "20:37:25.553 INFO  ProgressMeter - Starting traversal\n",
      "20:37:25.553 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "20:37:30.613 INFO  ProgressMeter -             MT:15967              0.1                501019        5944461.1\n",
      "20:37:30.613 INFO  ProgressMeter - Traversal complete. Processed 501019 total variants in 0.1 minutes.\n",
      "20:37:30.613 INFO  VariantsToTable - Shutting down engine\n",
      "[April 9, 2024, 8:37:30 PM MSK] org.broadinstitute.hellbender.tools.walkers.variantutils.VariantsToTable done. Elapsed time: 0.09 minutes.\n",
      "Runtime.totalMemory()=218103808\n",
      "/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/ipykernel_10295/4120375074.py:11: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  cv_dict[vcf] = pd.read_csv(f'{path_to_tables}/{vcf}.txt', delimiter='\\t', comment='#', skip_blank_lines=True)\n",
      "Using GATK jar /Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar VariantsToTable -V /Users/tanya/Documents/OTT/Conflicts_ClinVar/data/vcfs/clinvar_20200407.vcf -F CHROM -F POS -F ID -F REF -F ALT -F QUAL -F FILTER -F GENEINFO -F CLNVC -F CLNSIG -F CLNREVSTAT -F CLNSIGCONF -F CLNDN -F CLNDISDB -O /Users/tanya/Documents/OTT/Conflicts_ClinVar/tables/clinvar_20200407.txt\n",
      "20:37:32.901 INFO  NativeLibraryLoader - Loading libgkl_compression.dylib from jar:file:/Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.dylib\n",
      "20:37:32.935 WARN  NativeLibraryLoader - Unable to load libgkl_compression.dylib from native/libgkl_compression.dylib (/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression17325891260335653390.dylib: dlopen(/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression17325891260335653390.dylib, 0x0001): tried: '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression17325891260335653390.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression17325891260335653390.dylib' (no such file), '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression17325891260335653390.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')))\n",
      "20:37:32.938 INFO  NativeLibraryLoader - Loading libgkl_compression.dylib from jar:file:/Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.dylib\n",
      "20:37:32.941 WARN  NativeLibraryLoader - Unable to load libgkl_compression.dylib from native/libgkl_compression.dylib (/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression12486174413269750408.dylib: dlopen(/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression12486174413269750408.dylib, 0x0001): tried: '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression12486174413269750408.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression12486174413269750408.dylib' (no such file), '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression12486174413269750408.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')))\n",
      "20:37:33.016 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:37:33.018 INFO  VariantsToTable - The Genome Analysis Toolkit (GATK) v4.5.0.0\n",
      "20:37:33.018 INFO  VariantsToTable - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "20:37:33.019 INFO  VariantsToTable - Executing as tanya@Tatianas-MacBook-Air.local on Mac OS X v14.1.2 aarch64\n",
      "20:37:33.019 INFO  VariantsToTable - Java runtime: Java HotSpot(TM) 64-Bit Server VM v21+35-LTS-2513\n",
      "20:37:33.019 INFO  VariantsToTable - Start Date/Time: April 9, 2024, 8:37:32 PM MSK\n",
      "20:37:33.019 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:37:33.019 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:37:33.019 INFO  VariantsToTable - HTSJDK Version: 4.1.0\n",
      "20:37:33.019 INFO  VariantsToTable - Picard Version: 3.1.1\n",
      "20:37:33.020 INFO  VariantsToTable - Built for Spark Version: 3.5.0\n",
      "20:37:33.020 INFO  VariantsToTable - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "20:37:33.020 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "20:37:33.020 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "20:37:33.020 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "20:37:33.020 INFO  VariantsToTable - Deflater: JdkDeflater\n",
      "20:37:33.020 INFO  VariantsToTable - Inflater: JdkInflater\n",
      "20:37:33.021 INFO  VariantsToTable - GCS max retries/reopens: 20\n",
      "20:37:33.021 INFO  VariantsToTable - Requester pays: disabled\n",
      "20:37:33.021 INFO  VariantsToTable - Initializing engine\n",
      "20:37:33.097 INFO  FeatureManager - Using codec VCFCodec to read file file:///Users/tanya/Documents/OTT/Conflicts_ClinVar/data/vcfs/clinvar_20200407.vcf\n",
      "20:37:33.109 INFO  VariantsToTable - Done initializing engine\n",
      "20:37:33.110 WARN  VariantsToTable - Allele-specific fields will only be split if splitting multi-allelic variants is specified (`--split-multi-allelic` or `-SMA`\n",
      "20:37:33.110 INFO  ProgressMeter - Starting traversal\n",
      "20:37:33.110 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "20:37:39.596 INFO  ProgressMeter -             MT:15734              0.1                637120        5896529.4\n",
      "20:37:39.596 INFO  ProgressMeter - Traversal complete. Processed 637120 total variants in 0.1 minutes.\n",
      "20:37:39.596 INFO  VariantsToTable - Shutting down engine\n",
      "[April 9, 2024, 8:37:39 PM MSK] org.broadinstitute.hellbender.tools.walkers.variantutils.VariantsToTable done. Elapsed time: 0.11 minutes.\n",
      "Runtime.totalMemory()=219152384\n",
      "/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/ipykernel_10295/4120375074.py:11: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  cv_dict[vcf] = pd.read_csv(f'{path_to_tables}/{vcf}.txt', delimiter='\\t', comment='#', skip_blank_lines=True)\n",
      "Using GATK jar /Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar VariantsToTable -V /Users/tanya/Documents/OTT/Conflicts_ClinVar/data/vcfs/clinvar_20200927.vcf -F CHROM -F POS -F ID -F REF -F ALT -F QUAL -F FILTER -F GENEINFO -F CLNVC -F CLNSIG -F CLNREVSTAT -F CLNSIGCONF -F CLNDN -F CLNDISDB -O /Users/tanya/Documents/OTT/Conflicts_ClinVar/tables/clinvar_20200927.txt\n",
      "20:37:42.087 INFO  NativeLibraryLoader - Loading libgkl_compression.dylib from jar:file:/Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.dylib\n",
      "20:37:42.121 WARN  NativeLibraryLoader - Unable to load libgkl_compression.dylib from native/libgkl_compression.dylib (/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression11898060195704741896.dylib: dlopen(/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression11898060195704741896.dylib, 0x0001): tried: '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression11898060195704741896.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression11898060195704741896.dylib' (no such file), '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression11898060195704741896.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')))\n",
      "20:37:42.124 INFO  NativeLibraryLoader - Loading libgkl_compression.dylib from jar:file:/Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.dylib\n",
      "20:37:42.127 WARN  NativeLibraryLoader - Unable to load libgkl_compression.dylib from native/libgkl_compression.dylib (/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression3897019191460482073.dylib: dlopen(/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression3897019191460482073.dylib, 0x0001): tried: '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression3897019191460482073.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression3897019191460482073.dylib' (no such file), '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression3897019191460482073.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')))\n",
      "20:37:42.203 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:37:42.205 INFO  VariantsToTable - The Genome Analysis Toolkit (GATK) v4.5.0.0\n",
      "20:37:42.205 INFO  VariantsToTable - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "20:37:42.205 INFO  VariantsToTable - Executing as tanya@Tatianas-MacBook-Air.local on Mac OS X v14.1.2 aarch64\n",
      "20:37:42.205 INFO  VariantsToTable - Java runtime: Java HotSpot(TM) 64-Bit Server VM v21+35-LTS-2513\n",
      "20:37:42.205 INFO  VariantsToTable - Start Date/Time: April 9, 2024, 8:37:42 PM MSK\n",
      "20:37:42.205 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:37:42.205 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:37:42.206 INFO  VariantsToTable - HTSJDK Version: 4.1.0\n",
      "20:37:42.206 INFO  VariantsToTable - Picard Version: 3.1.1\n",
      "20:37:42.206 INFO  VariantsToTable - Built for Spark Version: 3.5.0\n",
      "20:37:42.206 INFO  VariantsToTable - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "20:37:42.206 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "20:37:42.206 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "20:37:42.207 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "20:37:42.207 INFO  VariantsToTable - Deflater: JdkDeflater\n",
      "20:37:42.207 INFO  VariantsToTable - Inflater: JdkInflater\n",
      "20:37:42.207 INFO  VariantsToTable - GCS max retries/reopens: 20\n",
      "20:37:42.207 INFO  VariantsToTable - Requester pays: disabled\n",
      "20:37:42.207 INFO  VariantsToTable - Initializing engine\n",
      "20:37:42.284 INFO  FeatureManager - Using codec VCFCodec to read file file:///Users/tanya/Documents/OTT/Conflicts_ClinVar/data/vcfs/clinvar_20200927.vcf\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "20:37:42.296 INFO  VariantsToTable - Done initializing engine\n",
      "20:37:42.297 WARN  VariantsToTable - Allele-specific fields will only be split if splitting multi-allelic variants is specified (`--split-multi-allelic` or `-SMA`\n",
      "20:37:42.297 INFO  ProgressMeter - Starting traversal\n",
      "20:37:42.297 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "20:37:50.842 INFO  ProgressMeter -             MT:14225              0.1                775501        5447209.1\n",
      "20:37:50.843 INFO  ProgressMeter - Traversal complete. Processed 775501 total variants in 0.1 minutes.\n",
      "20:37:50.843 INFO  VariantsToTable - Shutting down engine\n",
      "[April 9, 2024, 8:37:50 PM MSK] org.broadinstitute.hellbender.tools.walkers.variantutils.VariantsToTable done. Elapsed time: 0.15 minutes.\n",
      "Runtime.totalMemory()=219152384\n",
      "/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/ipykernel_10295/4120375074.py:11: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  cv_dict[vcf] = pd.read_csv(f'{path_to_tables}/{vcf}.txt', delimiter='\\t', comment='#', skip_blank_lines=True)\n",
      "Using GATK jar /Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar VariantsToTable -V /Users/tanya/Documents/OTT/Conflicts_ClinVar/data/vcfs/clinvar_20210328.vcf -F CHROM -F POS -F ID -F REF -F ALT -F QUAL -F FILTER -F GENEINFO -F CLNVC -F CLNSIG -F CLNREVSTAT -F CLNSIGCONF -F CLNDN -F CLNDISDB -O /Users/tanya/Documents/OTT/Conflicts_ClinVar/tables/clinvar_20210328.txt\n",
      "20:37:53.824 INFO  NativeLibraryLoader - Loading libgkl_compression.dylib from jar:file:/Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.dylib\n",
      "20:37:53.857 WARN  NativeLibraryLoader - Unable to load libgkl_compression.dylib from native/libgkl_compression.dylib (/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression9403170912775768202.dylib: dlopen(/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression9403170912775768202.dylib, 0x0001): tried: '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression9403170912775768202.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression9403170912775768202.dylib' (no such file), '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression9403170912775768202.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')))\n",
      "20:37:53.861 INFO  NativeLibraryLoader - Loading libgkl_compression.dylib from jar:file:/Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.dylib\n",
      "20:37:53.864 WARN  NativeLibraryLoader - Unable to load libgkl_compression.dylib from native/libgkl_compression.dylib (/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression5920218807523947238.dylib: dlopen(/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression5920218807523947238.dylib, 0x0001): tried: '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression5920218807523947238.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression5920218807523947238.dylib' (no such file), '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression5920218807523947238.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')))\n",
      "20:37:53.938 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:37:53.940 INFO  VariantsToTable - The Genome Analysis Toolkit (GATK) v4.5.0.0\n",
      "20:37:53.940 INFO  VariantsToTable - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "20:37:53.940 INFO  VariantsToTable - Executing as tanya@Tatianas-MacBook-Air.local on Mac OS X v14.1.2 aarch64\n",
      "20:37:53.940 INFO  VariantsToTable - Java runtime: Java HotSpot(TM) 64-Bit Server VM v21+35-LTS-2513\n",
      "20:37:53.940 INFO  VariantsToTable - Start Date/Time: April 9, 2024, 8:37:53 PM MSK\n",
      "20:37:53.940 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:37:53.941 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:37:53.941 INFO  VariantsToTable - HTSJDK Version: 4.1.0\n",
      "20:37:53.941 INFO  VariantsToTable - Picard Version: 3.1.1\n",
      "20:37:53.941 INFO  VariantsToTable - Built for Spark Version: 3.5.0\n",
      "20:37:53.941 INFO  VariantsToTable - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "20:37:53.941 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "20:37:53.942 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "20:37:53.942 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "20:37:53.942 INFO  VariantsToTable - Deflater: JdkDeflater\n",
      "20:37:53.942 INFO  VariantsToTable - Inflater: JdkInflater\n",
      "20:37:53.942 INFO  VariantsToTable - GCS max retries/reopens: 20\n",
      "20:37:53.942 INFO  VariantsToTable - Requester pays: disabled\n",
      "20:37:53.942 INFO  VariantsToTable - Initializing engine\n",
      "20:37:54.019 INFO  FeatureManager - Using codec VCFCodec to read file file:///Users/tanya/Documents/OTT/Conflicts_ClinVar/data/vcfs/clinvar_20210328.vcf\n",
      "20:37:54.030 INFO  VariantsToTable - Done initializing engine\n",
      "20:37:54.031 WARN  VariantsToTable - Allele-specific fields will only be split if splitting multi-allelic variants is specified (`--split-multi-allelic` or `-SMA`\n",
      "20:37:54.031 INFO  ProgressMeter - Starting traversal\n",
      "20:37:54.032 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "20:38:03.386 INFO  ProgressMeter -             MT:14924              0.2                840341        5391985.9\n",
      "20:38:03.387 INFO  ProgressMeter - Traversal complete. Processed 840341 total variants in 0.2 minutes.\n",
      "20:38:03.387 INFO  VariantsToTable - Shutting down engine\n",
      "[April 9, 2024, 8:38:03 PM MSK] org.broadinstitute.hellbender.tools.walkers.variantutils.VariantsToTable done. Elapsed time: 0.16 minutes.\n",
      "Runtime.totalMemory()=219152384\n",
      "/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/ipykernel_10295/4120375074.py:11: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  cv_dict[vcf] = pd.read_csv(f'{path_to_tables}/{vcf}.txt', delimiter='\\t', comment='#', skip_blank_lines=True)\n",
      "Using GATK jar /Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar VariantsToTable -V /Users/tanya/Documents/OTT/Conflicts_ClinVar/data/vcfs/clinvar_20210929.vcf -F CHROM -F POS -F ID -F REF -F ALT -F QUAL -F FILTER -F GENEINFO -F CLNVC -F CLNSIG -F CLNREVSTAT -F CLNSIGCONF -F CLNDN -F CLNDISDB -O /Users/tanya/Documents/OTT/Conflicts_ClinVar/tables/clinvar_20210929.txt\n",
      "20:38:06.544 INFO  NativeLibraryLoader - Loading libgkl_compression.dylib from jar:file:/Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.dylib\n",
      "20:38:06.578 WARN  NativeLibraryLoader - Unable to load libgkl_compression.dylib from native/libgkl_compression.dylib (/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression8081317950055094393.dylib: dlopen(/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression8081317950055094393.dylib, 0x0001): tried: '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression8081317950055094393.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression8081317950055094393.dylib' (no such file), '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression8081317950055094393.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')))\n",
      "20:38:06.581 INFO  NativeLibraryLoader - Loading libgkl_compression.dylib from jar:file:/Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.dylib\n",
      "20:38:06.584 WARN  NativeLibraryLoader - Unable to load libgkl_compression.dylib from native/libgkl_compression.dylib (/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression11796246342145034378.dylib: dlopen(/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression11796246342145034378.dylib, 0x0001): tried: '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression11796246342145034378.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression11796246342145034378.dylib' (no such file), '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression11796246342145034378.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')))\n",
      "20:38:06.658 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:38:06.660 INFO  VariantsToTable - The Genome Analysis Toolkit (GATK) v4.5.0.0\n",
      "20:38:06.660 INFO  VariantsToTable - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "20:38:06.660 INFO  VariantsToTable - Executing as tanya@Tatianas-MacBook-Air.local on Mac OS X v14.1.2 aarch64\n",
      "20:38:06.660 INFO  VariantsToTable - Java runtime: Java HotSpot(TM) 64-Bit Server VM v21+35-LTS-2513\n",
      "20:38:06.660 INFO  VariantsToTable - Start Date/Time: April 9, 2024, 8:38:06 PM MSK\n",
      "20:38:06.660 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:38:06.660 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:38:06.661 INFO  VariantsToTable - HTSJDK Version: 4.1.0\n",
      "20:38:06.661 INFO  VariantsToTable - Picard Version: 3.1.1\n",
      "20:38:06.661 INFO  VariantsToTable - Built for Spark Version: 3.5.0\n",
      "20:38:06.661 INFO  VariantsToTable - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "20:38:06.661 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "20:38:06.661 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "20:38:06.661 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "20:38:06.661 INFO  VariantsToTable - Deflater: JdkDeflater\n",
      "20:38:06.662 INFO  VariantsToTable - Inflater: JdkInflater\n",
      "20:38:06.662 INFO  VariantsToTable - GCS max retries/reopens: 20\n",
      "20:38:06.662 INFO  VariantsToTable - Requester pays: disabled\n",
      "20:38:06.662 INFO  VariantsToTable - Initializing engine\n",
      "20:38:06.740 INFO  FeatureManager - Using codec VCFCodec to read file file:///Users/tanya/Documents/OTT/Conflicts_ClinVar/data/vcfs/clinvar_20210929.vcf\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "20:38:06.751 INFO  VariantsToTable - Done initializing engine\n",
      "20:38:06.752 WARN  VariantsToTable - Allele-specific fields will only be split if splitting multi-allelic variants is specified (`--split-multi-allelic` or `-SMA`\n",
      "20:38:06.752 INFO  ProgressMeter - Starting traversal\n",
      "20:38:06.752 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "20:38:16.762 INFO  ProgressMeter -          19:10986193              0.2                940000        5636618.0\n",
      "20:38:18.265 INFO  ProgressMeter -             MT:14663              0.2               1079418        5625875.6\n",
      "20:38:18.265 INFO  ProgressMeter - Traversal complete. Processed 1079418 total variants in 0.2 minutes.\n",
      "20:38:18.265 INFO  VariantsToTable - Shutting down engine\n",
      "[April 9, 2024, 8:38:18 PM MSK] org.broadinstitute.hellbender.tools.walkers.variantutils.VariantsToTable done. Elapsed time: 0.20 minutes.\n",
      "Runtime.totalMemory()=226492416\n",
      "/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/ipykernel_10295/4120375074.py:11: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  cv_dict[vcf] = pd.read_csv(f'{path_to_tables}/{vcf}.txt', delimiter='\\t', comment='#', skip_blank_lines=True)\n",
      "Using GATK jar /Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar VariantsToTable -V /Users/tanya/Documents/OTT/Conflicts_ClinVar/data/vcfs/clinvar_20220403.vcf -F CHROM -F POS -F ID -F REF -F ALT -F QUAL -F FILTER -F GENEINFO -F CLNVC -F CLNSIG -F CLNREVSTAT -F CLNSIGCONF -F CLNDN -F CLNDISDB -O /Users/tanya/Documents/OTT/Conflicts_ClinVar/tables/clinvar_20220403.txt\n",
      "20:38:21.920 INFO  NativeLibraryLoader - Loading libgkl_compression.dylib from jar:file:/Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.dylib\n",
      "20:38:21.954 WARN  NativeLibraryLoader - Unable to load libgkl_compression.dylib from native/libgkl_compression.dylib (/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression424894755622957639.dylib: dlopen(/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression424894755622957639.dylib, 0x0001): tried: '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression424894755622957639.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression424894755622957639.dylib' (no such file), '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression424894755622957639.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')))\n",
      "20:38:21.957 INFO  NativeLibraryLoader - Loading libgkl_compression.dylib from jar:file:/Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.dylib\n",
      "20:38:21.960 WARN  NativeLibraryLoader - Unable to load libgkl_compression.dylib from native/libgkl_compression.dylib (/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression15891473998179473028.dylib: dlopen(/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression15891473998179473028.dylib, 0x0001): tried: '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression15891473998179473028.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression15891473998179473028.dylib' (no such file), '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression15891473998179473028.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')))\n",
      "20:38:22.034 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:38:22.036 INFO  VariantsToTable - The Genome Analysis Toolkit (GATK) v4.5.0.0\n",
      "20:38:22.036 INFO  VariantsToTable - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "20:38:22.036 INFO  VariantsToTable - Executing as tanya@Tatianas-MacBook-Air.local on Mac OS X v14.1.2 aarch64\n",
      "20:38:22.036 INFO  VariantsToTable - Java runtime: Java HotSpot(TM) 64-Bit Server VM v21+35-LTS-2513\n",
      "20:38:22.036 INFO  VariantsToTable - Start Date/Time: April 9, 2024, 8:38:21 PM MSK\n",
      "20:38:22.037 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:38:22.037 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:38:22.037 INFO  VariantsToTable - HTSJDK Version: 4.1.0\n",
      "20:38:22.037 INFO  VariantsToTable - Picard Version: 3.1.1\n",
      "20:38:22.037 INFO  VariantsToTable - Built for Spark Version: 3.5.0\n",
      "20:38:22.037 INFO  VariantsToTable - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "20:38:22.038 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "20:38:22.038 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "20:38:22.038 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "20:38:22.038 INFO  VariantsToTable - Deflater: JdkDeflater\n",
      "20:38:22.038 INFO  VariantsToTable - Inflater: JdkInflater\n",
      "20:38:22.038 INFO  VariantsToTable - GCS max retries/reopens: 20\n",
      "20:38:22.038 INFO  VariantsToTable - Requester pays: disabled\n",
      "20:38:22.038 INFO  VariantsToTable - Initializing engine\n",
      "20:38:22.113 INFO  FeatureManager - Using codec VCFCodec to read file file:///Users/tanya/Documents/OTT/Conflicts_ClinVar/data/vcfs/clinvar_20220403.vcf\n",
      "20:38:22.125 INFO  VariantsToTable - Done initializing engine\n",
      "20:38:22.126 WARN  VariantsToTable - Allele-specific fields will only be split if splitting multi-allelic variants is specified (`--split-multi-allelic` or `-SMA`\n",
      "20:38:22.126 INFO  ProgressMeter - Starting traversal\n",
      "20:38:22.126 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "20:38:32.133 INFO  ProgressMeter -          16:74719181              0.2                996000        5974207.7\n",
      "20:38:35.191 INFO  ProgressMeter -             MT:14002              0.2               1303558        5986489.1\n",
      "20:38:35.191 INFO  ProgressMeter - Traversal complete. Processed 1303558 total variants in 0.2 minutes.\n",
      "20:38:35.191 INFO  VariantsToTable - Shutting down engine\n",
      "[April 9, 2024, 8:38:35 PM MSK] org.broadinstitute.hellbender.tools.walkers.variantutils.VariantsToTable done. Elapsed time: 0.22 minutes.\n",
      "Runtime.totalMemory()=219152384\n",
      "/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/ipykernel_10295/4120375074.py:11: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  cv_dict[vcf] = pd.read_csv(f'{path_to_tables}/{vcf}.txt', delimiter='\\t', comment='#', skip_blank_lines=True)\n",
      "Using GATK jar /Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar VariantsToTable -V /Users/tanya/Documents/OTT/Conflicts_ClinVar/data/vcfs/clinvar_20221001.vcf -F CHROM -F POS -F ID -F REF -F ALT -F QUAL -F FILTER -F GENEINFO -F CLNVC -F CLNSIG -F CLNREVSTAT -F CLNSIGCONF -F CLNDN -F CLNDISDB -O /Users/tanya/Documents/OTT/Conflicts_ClinVar/tables/clinvar_20221001.txt\n",
      "20:38:39.345 INFO  NativeLibraryLoader - Loading libgkl_compression.dylib from jar:file:/Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.dylib\n",
      "20:38:39.380 WARN  NativeLibraryLoader - Unable to load libgkl_compression.dylib from native/libgkl_compression.dylib (/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression18434496083749950200.dylib: dlopen(/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression18434496083749950200.dylib, 0x0001): tried: '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression18434496083749950200.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression18434496083749950200.dylib' (no such file), '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression18434496083749950200.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')))\n",
      "20:38:39.383 INFO  NativeLibraryLoader - Loading libgkl_compression.dylib from jar:file:/Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.dylib\n",
      "20:38:39.386 WARN  NativeLibraryLoader - Unable to load libgkl_compression.dylib from native/libgkl_compression.dylib (/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression12901516121662099522.dylib: dlopen(/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression12901516121662099522.dylib, 0x0001): tried: '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression12901516121662099522.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression12901516121662099522.dylib' (no such file), '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression12901516121662099522.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')))\n",
      "20:38:39.460 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:38:39.462 INFO  VariantsToTable - The Genome Analysis Toolkit (GATK) v4.5.0.0\n",
      "20:38:39.462 INFO  VariantsToTable - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "20:38:39.463 INFO  VariantsToTable - Executing as tanya@Tatianas-MacBook-Air.local on Mac OS X v14.1.2 aarch64\n",
      "20:38:39.463 INFO  VariantsToTable - Java runtime: Java HotSpot(TM) 64-Bit Server VM v21+35-LTS-2513\n",
      "20:38:39.463 INFO  VariantsToTable - Start Date/Time: April 9, 2024, 8:38:39 PM MSK\n",
      "20:38:39.463 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:38:39.463 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:38:39.463 INFO  VariantsToTable - HTSJDK Version: 4.1.0\n",
      "20:38:39.463 INFO  VariantsToTable - Picard Version: 3.1.1\n",
      "20:38:39.464 INFO  VariantsToTable - Built for Spark Version: 3.5.0\n",
      "20:38:39.464 INFO  VariantsToTable - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "20:38:39.464 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "20:38:39.464 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "20:38:39.464 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "20:38:39.464 INFO  VariantsToTable - Deflater: JdkDeflater\n",
      "20:38:39.464 INFO  VariantsToTable - Inflater: JdkInflater\n",
      "20:38:39.465 INFO  VariantsToTable - GCS max retries/reopens: 20\n",
      "20:38:39.465 INFO  VariantsToTable - Requester pays: disabled\n",
      "20:38:39.465 INFO  VariantsToTable - Initializing engine\n",
      "20:38:39.542 INFO  FeatureManager - Using codec VCFCodec to read file file:///Users/tanya/Documents/OTT/Conflicts_ClinVar/data/vcfs/clinvar_20221001.vcf\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "20:38:39.555 INFO  VariantsToTable - Done initializing engine\n",
      "20:38:39.556 WARN  VariantsToTable - Allele-specific fields will only be split if splitting multi-allelic variants is specified (`--split-multi-allelic` or `-SMA`\n",
      "20:38:39.556 INFO  ProgressMeter - Starting traversal\n",
      "20:38:39.557 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "20:38:47.512 INFO  VariantsToTable - Shutting down engine\n",
      "[April 9, 2024, 8:38:47 PM MSK] org.broadinstitute.hellbender.tools.walkers.variantutils.VariantsToTable done. Elapsed time: 0.14 minutes.\n",
      "Runtime.totalMemory()=219152384\n",
      "htsjdk.tribble.TribbleException: The provided VCF file is malformed at approximately line number 789412: unparsable vcf record with allele YT, for input source: /Users/tanya/Documents/OTT/Conflicts_ClinVar/data/vcfs/clinvar_20221001.vcf\n",
      "\tat htsjdk.variant.vcf.AbstractVCFCodec.generateException(AbstractVCFCodec.java:887)\n",
      "\tat htsjdk.variant.vcf.AbstractVCFCodec.checkAllele(AbstractVCFCodec.java:678)\n",
      "\tat htsjdk.variant.vcf.AbstractVCFCodec.parseSingleAltAllele(AbstractVCFCodec.java:706)\n",
      "\tat htsjdk.variant.vcf.AbstractVCFCodec.parseAlleles(AbstractVCFCodec.java:645)\n",
      "\tat htsjdk.variant.vcf.AbstractVCFCodec.parseVCFLine(AbstractVCFCodec.java:443)\n",
      "\tat htsjdk.variant.vcf.AbstractVCFCodec.decodeLine(AbstractVCFCodec.java:384)\n",
      "\tat htsjdk.variant.vcf.AbstractVCFCodec.decode(AbstractVCFCodec.java:328)\n",
      "\tat htsjdk.variant.vcf.AbstractVCFCodec.decode(AbstractVCFCodec.java:48)\n",
      "\tat htsjdk.tribble.AsciiFeatureCodec.decode(AsciiFeatureCodec.java:70)\n",
      "\tat htsjdk.tribble.AsciiFeatureCodec.decode(AsciiFeatureCodec.java:37)\n",
      "\tat htsjdk.tribble.TribbleIndexedFeatureReader$WFIterator.readNextRecord(TribbleIndexedFeatureReader.java:377)\n",
      "\tat htsjdk.tribble.TribbleIndexedFeatureReader$WFIterator.next(TribbleIndexedFeatureReader.java:356)\n",
      "\tat htsjdk.tribble.TribbleIndexedFeatureReader$WFIterator.next(TribbleIndexedFeatureReader.java:317)\n",
      "\tat java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)\n",
      "\tat java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1939)\n",
      "\tat java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)\n",
      "\tat java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)\n",
      "\tat java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)\n",
      "\tat java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)\n",
      "\tat java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)\n",
      "\tat java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)\n",
      "\tat org.broadinstitute.hellbender.engine.VariantWalker.traverse(VariantWalker.java:102)\n",
      "\tat org.broadinstitute.hellbender.engine.GATKTool.doWork(GATKTool.java:1098)\n",
      "\tat org.broadinstitute.hellbender.cmdline.CommandLineProgram.runTool(CommandLineProgram.java:149)\n",
      "\tat org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMainPostParseArgs(CommandLineProgram.java:198)\n",
      "\tat org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMain(CommandLineProgram.java:217)\n",
      "\tat org.broadinstitute.hellbender.Main.runCommandLineProgram(Main.java:166)\n",
      "\tat org.broadinstitute.hellbender.Main.mainEntry(Main.java:209)\n",
      "\tat org.broadinstitute.hellbender.Main.main(Main.java:306)\n",
      "Using GATK jar /Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar VariantsToTable -V /Users/tanya/Documents/OTT/Conflicts_ClinVar/data/vcfs/clinvar_20230404.vcf -F CHROM -F POS -F ID -F REF -F ALT -F QUAL -F FILTER -F GENEINFO -F CLNVC -F CLNSIG -F CLNREVSTAT -F CLNSIGCONF -F CLNDN -F CLNDISDB -O /Users/tanya/Documents/OTT/Conflicts_ClinVar/tables/clinvar_20230404.txt\n",
      "20:38:50.515 INFO  NativeLibraryLoader - Loading libgkl_compression.dylib from jar:file:/Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.dylib\n",
      "20:38:50.549 WARN  NativeLibraryLoader - Unable to load libgkl_compression.dylib from native/libgkl_compression.dylib (/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression2977140779474336961.dylib: dlopen(/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression2977140779474336961.dylib, 0x0001): tried: '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression2977140779474336961.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression2977140779474336961.dylib' (no such file), '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression2977140779474336961.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')))\n",
      "20:38:50.553 INFO  NativeLibraryLoader - Loading libgkl_compression.dylib from jar:file:/Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.dylib\n",
      "20:38:50.556 WARN  NativeLibraryLoader - Unable to load libgkl_compression.dylib from native/libgkl_compression.dylib (/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression16511955548368712706.dylib: dlopen(/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression16511955548368712706.dylib, 0x0001): tried: '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression16511955548368712706.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression16511955548368712706.dylib' (no such file), '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression16511955548368712706.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')))\n",
      "20:38:50.631 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:38:50.633 INFO  VariantsToTable - The Genome Analysis Toolkit (GATK) v4.5.0.0\n",
      "20:38:50.633 INFO  VariantsToTable - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "20:38:50.633 INFO  VariantsToTable - Executing as tanya@Tatianas-MacBook-Air.local on Mac OS X v14.1.2 aarch64\n",
      "20:38:50.634 INFO  VariantsToTable - Java runtime: Java HotSpot(TM) 64-Bit Server VM v21+35-LTS-2513\n",
      "20:38:50.634 INFO  VariantsToTable - Start Date/Time: April 9, 2024, 8:38:50 PM MSK\n",
      "20:38:50.634 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:38:50.634 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:38:50.634 INFO  VariantsToTable - HTSJDK Version: 4.1.0\n",
      "20:38:50.634 INFO  VariantsToTable - Picard Version: 3.1.1\n",
      "20:38:50.634 INFO  VariantsToTable - Built for Spark Version: 3.5.0\n",
      "20:38:50.635 INFO  VariantsToTable - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "20:38:50.635 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "20:38:50.635 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "20:38:50.635 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "20:38:50.635 INFO  VariantsToTable - Deflater: JdkDeflater\n",
      "20:38:50.635 INFO  VariantsToTable - Inflater: JdkInflater\n",
      "20:38:50.635 INFO  VariantsToTable - GCS max retries/reopens: 20\n",
      "20:38:50.635 INFO  VariantsToTable - Requester pays: disabled\n",
      "20:38:50.636 INFO  VariantsToTable - Initializing engine\n",
      "20:38:50.713 INFO  FeatureManager - Using codec VCFCodec to read file file:///Users/tanya/Documents/OTT/Conflicts_ClinVar/data/vcfs/clinvar_20230404.vcf\n",
      "20:38:50.724 INFO  VariantsToTable - Done initializing engine\n",
      "20:38:50.725 WARN  VariantsToTable - Allele-specific fields will only be split if splitting multi-allelic variants is specified (`--split-multi-allelic` or `-SMA`\n",
      "20:38:50.725 INFO  ProgressMeter - Starting traversal\n",
      "20:38:50.726 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "20:39:00.731 INFO  ProgressMeter -          9:101535461              0.2                998000        5986802.6\n",
      "20:39:10.731 INFO  ProgressMeter -          21:46366702              0.3               1993000        5977505.6\n",
      "20:39:12.053 INFO  ProgressMeter -             MT:15059              0.4               2122324        5970808.8\n",
      "20:39:12.053 INFO  ProgressMeter - Traversal complete. Processed 2122324 total variants in 0.4 minutes.\n",
      "20:39:12.053 INFO  VariantsToTable - Shutting down engine\n",
      "[April 9, 2024, 8:39:12 PM MSK] org.broadinstitute.hellbender.tools.walkers.variantutils.VariantsToTable done. Elapsed time: 0.36 minutes.\n",
      "Runtime.totalMemory()=217055232\n",
      "/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/ipykernel_10295/4120375074.py:11: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  cv_dict[vcf] = pd.read_csv(f'{path_to_tables}/{vcf}.txt', delimiter='\\t', comment='#', skip_blank_lines=True)\n",
      "Using GATK jar /Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar VariantsToTable -V /Users/tanya/Documents/OTT/Conflicts_ClinVar/data/vcfs/clinvar_20230930.vcf -F CHROM -F POS -F ID -F REF -F ALT -F QUAL -F FILTER -F GENEINFO -F CLNVC -F CLNSIG -F CLNREVSTAT -F CLNSIGCONF -F CLNDN -F CLNDISDB -O /Users/tanya/Documents/OTT/Conflicts_ClinVar/tables/clinvar_20230930.txt\n",
      "20:39:18.010 INFO  NativeLibraryLoader - Loading libgkl_compression.dylib from jar:file:/Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.dylib\n",
      "20:39:18.048 WARN  NativeLibraryLoader - Unable to load libgkl_compression.dylib from native/libgkl_compression.dylib (/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression6333446972728739461.dylib: dlopen(/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression6333446972728739461.dylib, 0x0001): tried: '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression6333446972728739461.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression6333446972728739461.dylib' (no such file), '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression6333446972728739461.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')))\n",
      "20:39:18.051 INFO  NativeLibraryLoader - Loading libgkl_compression.dylib from jar:file:/Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.dylib\n",
      "20:39:18.055 WARN  NativeLibraryLoader - Unable to load libgkl_compression.dylib from native/libgkl_compression.dylib (/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression17470721834184705610.dylib: dlopen(/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression17470721834184705610.dylib, 0x0001): tried: '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression17470721834184705610.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression17470721834184705610.dylib' (no such file), '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression17470721834184705610.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')))\n",
      "20:39:18.136 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:39:18.138 INFO  VariantsToTable - The Genome Analysis Toolkit (GATK) v4.5.0.0\n",
      "20:39:18.138 INFO  VariantsToTable - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "20:39:18.138 INFO  VariantsToTable - Executing as tanya@Tatianas-MacBook-Air.local on Mac OS X v14.1.2 aarch64\n",
      "20:39:18.138 INFO  VariantsToTable - Java runtime: Java HotSpot(TM) 64-Bit Server VM v21+35-LTS-2513\n",
      "20:39:18.138 INFO  VariantsToTable - Start Date/Time: April 9, 2024, 8:39:17 PM MSK\n",
      "20:39:18.138 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:39:18.139 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:39:18.139 INFO  VariantsToTable - HTSJDK Version: 4.1.0\n",
      "20:39:18.139 INFO  VariantsToTable - Picard Version: 3.1.1\n",
      "20:39:18.139 INFO  VariantsToTable - Built for Spark Version: 3.5.0\n",
      "20:39:18.139 INFO  VariantsToTable - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "20:39:18.139 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "20:39:18.140 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "20:39:18.140 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "20:39:18.140 INFO  VariantsToTable - Deflater: JdkDeflater\n",
      "20:39:18.140 INFO  VariantsToTable - Inflater: JdkInflater\n",
      "20:39:18.140 INFO  VariantsToTable - GCS max retries/reopens: 20\n",
      "20:39:18.140 INFO  VariantsToTable - Requester pays: disabled\n",
      "20:39:18.140 INFO  VariantsToTable - Initializing engine\n",
      "20:39:18.219 INFO  FeatureManager - Using codec VCFCodec to read file file:///Users/tanya/Documents/OTT/Conflicts_ClinVar/data/vcfs/clinvar_20230930.vcf\n",
      "20:39:18.233 INFO  VariantsToTable - Done initializing engine\n",
      "20:39:18.234 WARN  VariantsToTable - Allele-specific fields will only be split if splitting multi-allelic variants is specified (`--split-multi-allelic` or `-SMA`\n",
      "20:39:18.234 INFO  ProgressMeter - Starting traversal\n",
      "20:39:18.234 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "20:39:28.244 INFO  ProgressMeter -          8:143920736              0.2                999000        5989807.1\n",
      "20:39:38.241 INFO  ProgressMeter -          19:11252177              0.3               1968000        5901934.3\n",
      "20:39:41.096 INFO  ProgressMeter -             MT:15650              0.4               2255167        5918555.7\n",
      "20:39:41.097 INFO  ProgressMeter - Traversal complete. Processed 2255167 total variants in 0.4 minutes.\n",
      "20:39:41.097 INFO  VariantsToTable - Shutting down engine\n",
      "[April 9, 2024, 8:39:41 PM MSK] org.broadinstitute.hellbender.tools.walkers.variantutils.VariantsToTable done. Elapsed time: 0.39 minutes.\n",
      "Runtime.totalMemory()=217055232\n",
      "/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/ipykernel_10295/4120375074.py:11: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  cv_dict[vcf] = pd.read_csv(f'{path_to_tables}/{vcf}.txt', delimiter='\\t', comment='#', skip_blank_lines=True)\n",
      "Using GATK jar /Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar VariantsToTable -V /Users/tanya/Documents/OTT/Conflicts_ClinVar/data/vcfs/clinvar_20240407.vcf -F CHROM -F POS -F ID -F REF -F ALT -F QUAL -F FILTER -F GENEINFO -F CLNVC -F CLNSIG -F CLNREVSTAT -F CLNSIGCONF -F CLNDN -F CLNDISDB -O /Users/tanya/Documents/OTT/Conflicts_ClinVar/tables/clinvar_20240407.txt\n",
      "20:39:47.401 INFO  NativeLibraryLoader - Loading libgkl_compression.dylib from jar:file:/Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.dylib\n",
      "20:39:47.442 WARN  NativeLibraryLoader - Unable to load libgkl_compression.dylib from native/libgkl_compression.dylib (/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression5290024935093666587.dylib: dlopen(/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression5290024935093666587.dylib, 0x0001): tried: '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression5290024935093666587.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression5290024935093666587.dylib' (no such file), '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression5290024935093666587.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')))\n",
      "20:39:47.445 INFO  NativeLibraryLoader - Loading libgkl_compression.dylib from jar:file:/Users/tanya/Documents/tools/gatk-4.5.0.0/gatk-package-4.5.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.dylib\n",
      "20:39:47.449 WARN  NativeLibraryLoader - Unable to load libgkl_compression.dylib from native/libgkl_compression.dylib (/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression13624020055192298395.dylib: dlopen(/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression13624020055192298395.dylib, 0x0001): tried: '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression13624020055192298395.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression13624020055192298395.dylib' (no such file), '/private/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/libgkl_compression13624020055192298395.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')))\n",
      "20:39:47.531 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:39:47.534 INFO  VariantsToTable - The Genome Analysis Toolkit (GATK) v4.5.0.0\n",
      "20:39:47.534 INFO  VariantsToTable - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "20:39:47.534 INFO  VariantsToTable - Executing as tanya@Tatianas-MacBook-Air.local on Mac OS X v14.1.2 aarch64\n",
      "20:39:47.534 INFO  VariantsToTable - Java runtime: Java HotSpot(TM) 64-Bit Server VM v21+35-LTS-2513\n",
      "20:39:47.534 INFO  VariantsToTable - Start Date/Time: April 9, 2024, 8:39:47 PM MSK\n",
      "20:39:47.534 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:39:47.534 INFO  VariantsToTable - ------------------------------------------------------------\n",
      "20:39:47.535 INFO  VariantsToTable - HTSJDK Version: 4.1.0\n",
      "20:39:47.535 INFO  VariantsToTable - Picard Version: 3.1.1\n",
      "20:39:47.535 INFO  VariantsToTable - Built for Spark Version: 3.5.0\n",
      "20:39:47.535 INFO  VariantsToTable - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "20:39:47.535 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "20:39:47.535 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "20:39:47.535 INFO  VariantsToTable - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "20:39:47.535 INFO  VariantsToTable - Deflater: JdkDeflater\n",
      "20:39:47.536 INFO  VariantsToTable - Inflater: JdkInflater\n",
      "20:39:47.536 INFO  VariantsToTable - GCS max retries/reopens: 20\n",
      "20:39:47.536 INFO  VariantsToTable - Requester pays: disabled\n",
      "20:39:47.536 INFO  VariantsToTable - Initializing engine\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "20:39:47.615 INFO  FeatureManager - Using codec VCFCodec to read file file:///Users/tanya/Documents/OTT/Conflicts_ClinVar/data/vcfs/clinvar_20240407.vcf\n",
      "20:39:47.629 INFO  VariantsToTable - Done initializing engine\n",
      "20:39:47.629 WARN  VariantsToTable - Allele-specific fields will only be split if splitting multi-allelic variants is specified (`--split-multi-allelic` or `-SMA`\n",
      "20:39:47.630 INFO  ProgressMeter - Starting traversal\n",
      "20:39:47.630 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "20:39:57.637 INFO  ProgressMeter -           7:41972521              0.2               1027000        6159536.2\n",
      "20:40:07.635 INFO  ProgressMeter -          16:27442995              0.3               2033000        6097780.4\n",
      "20:40:14.504 INFO  ProgressMeter -             MT:15434              0.4               2726235        6086704.6\n",
      "20:40:14.504 INFO  ProgressMeter - Traversal complete. Processed 2726235 total variants in 0.4 minutes.\n",
      "20:40:14.505 INFO  VariantsToTable - Shutting down engine\n",
      "[April 9, 2024, 8:40:14 PM MSK] org.broadinstitute.hellbender.tools.walkers.variantutils.VariantsToTable done. Elapsed time: 0.45 minutes.\n",
      "Runtime.totalMemory()=261095424\n",
      "/var/folders/sy/0s6mz48x25xdw7n08d30rp5w0000gn/T/ipykernel_10295/4120375074.py:11: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  cv_dict[vcf] = pd.read_csv(f'{path_to_tables}/{vcf}.txt', delimiter='\\t', comment='#', skip_blank_lines=True)\n"
     ]
    }
   ],
   "source": [
    "# Transform VCFs to tables\n",
    "\n",
    "cv_dict = {}   # dictionary where keys are VCFs names and items are pandas dataframes contains collumns of interest from VCF\n",
    "\n",
    "\n",
    "for vcf in vcfs:   # select INFO fields\n",
    "    os.system(f'{path_to_gatk} VariantsToTable -V {path_to_vcfs}/{vcf}.vcf \\\n",
    "    -F CHROM -F POS -F ID -F REF -F ALT -F QUAL -F FILTER  -F GENEINFO -F CLNVC -F CLNSIG \\\n",
    "    -F CLNREVSTAT -F CLNSIGCONF -F CLNDN  -F CLNDISDB -O {path_to_tables}/{vcf}.txt')\n",
    "    \n",
    "    cv_dict[vcf] = pd.read_csv(f'{path_to_tables}/{vcf}.txt', delimiter='\\t', comment='#', skip_blank_lines=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "3963b214",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load tables, if it already exists\n",
    "\n",
    "cv_dict = {} \n",
    "\n",
    "for vcf in vcfs:\n",
    "    cv_dict[vcf] = pd.read_csv(f'{path_to_tables}/{vcf}.txt', delimiter='\\t', comment='#', \n",
    "                               skip_blank_lines=True, low_memory=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "bced4758",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Transform columns data\n",
    "\n",
    "for vcf in cv_dict:   # remove numerical GENEINFO data, save only gene symbol\n",
    "    cv_dict[vcf][[f'GENEINFO']] = cv_dict[vcf][[f'GENEINFO']].replace(to_replace=':.*', regex=True, value='')\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "edff6588",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>CLNSIG_MASK</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Affects</td>\n",
       "      <td>Affects</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Affects,_association</td>\n",
       "      <td>Affects</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 CLNSIG CLNSIG_MASK\n",
       "0               Affects     Affects\n",
       "1  Affects,_association     Affects"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clnsig_mask = pd.read_csv(f'{path_to_data}/CLNSIG_MASK.csv', sep=',')\n",
    "clnsig_mask.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "f6f318c4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Apply mask for clnsig column\n",
    "\n",
    "for vcf in cv_dict:\n",
    "    cv_dict[vcf]['CLNSIG_MASK'] =  cv_dict[vcf]['CLNSIG'].map(clnsig_mask.set_index('CLNSIG')['CLNSIG_MASK'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "cf68e53e",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(f'{path_to_temp}/cv_dict.pkl', 'wb') as file:   # save  dict\n",
    "    pickle.dump(cv_dict, file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "e1a70589",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>GENEINFO</th>\n",
       "      <th>CLNVC</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>CLNREVSTAT</th>\n",
       "      <th>CLNSIGCONF</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>CLNDISDB</th>\n",
       "      <th>CLNSIG_MASK</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>1</td>\n",
       "      <td>930200</td>\n",
       "      <td>1043045</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>Conflicting_classifications_of_pathogenicity</td>\n",
       "      <td>criteria_provided,_conflicting_classifications</td>\n",
       "      <td>Uncertain_significance(1)|Likely_benign(1)</td>\n",
       "      <td>not_provided|Inborn_genetic_diseases</td>\n",
       "      <td>MedGen:CN517202|MeSH:D030342,MedGen:C0950123</td>\n",
       "      <td>COI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>1</td>\n",
       "      <td>935839</td>\n",
       "      <td>1085785</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>Conflicting_classifications_of_pathogenicity</td>\n",
       "      <td>criteria_provided,_conflicting_classifications</td>\n",
       "      <td>Uncertain_significance(1)|Likely_benign(1)</td>\n",
       "      <td>Inborn_genetic_diseases|not_provided</td>\n",
       "      <td>MeSH:D030342,MedGen:C0950123|MedGen:C3661900</td>\n",
       "      <td>COI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>214</th>\n",
       "      <td>1</td>\n",
       "      <td>939117</td>\n",
       "      <td>1427749</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>Conflicting_classifications_of_pathogenicity</td>\n",
       "      <td>criteria_provided,_conflicting_classifications</td>\n",
       "      <td>Uncertain_significance(1)|Likely_benign(1)</td>\n",
       "      <td>not_provided|Inborn_genetic_diseases</td>\n",
       "      <td>MedGen:C3661900|MeSH:D030342,MedGen:C0950123</td>\n",
       "      <td>COI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>325</th>\n",
       "      <td>1</td>\n",
       "      <td>941284</td>\n",
       "      <td>1006858</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>Conflicting_classifications_of_pathogenicity</td>\n",
       "      <td>criteria_provided,_conflicting_classifications</td>\n",
       "      <td>Uncertain_significance(1)|Likely_benign(1)</td>\n",
       "      <td>not_specified|not_provided</td>\n",
       "      <td>MedGen:CN169374|MedGen:C3661900</td>\n",
       "      <td>COI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>438</th>\n",
       "      <td>1</td>\n",
       "      <td>942470</td>\n",
       "      <td>2334732</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>Conflicting_classifications_of_pathogenicity</td>\n",
       "      <td>criteria_provided,_conflicting_classifications</td>\n",
       "      <td>Uncertain_significance(1)|Likely_benign(1)</td>\n",
       "      <td>not_provided|Inborn_genetic_diseases</td>\n",
       "      <td>MedGen:C3661900|MeSH:D030342,MedGen:C0950123</td>\n",
       "      <td>COI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2725349</th>\n",
       "      <td>MT</td>\n",
       "      <td>12278</td>\n",
       "      <td>690182</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>MT-TL2</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>Conflicting_classifications_of_pathogenicity</td>\n",
       "      <td>criteria_provided,_conflicting_classifications</td>\n",
       "      <td>Likely_pathogenic(1)|Uncertain_significance(2)</td>\n",
       "      <td>Juvenile_myopathy,_encephalopathy,_lactic_acid...</td>\n",
       "      <td>MONDO:MONDO:0010789,MedGen:C0162671,OMIM:54000...</td>\n",
       "      <td>COI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2725384</th>\n",
       "      <td>MT</td>\n",
       "      <td>12372</td>\n",
       "      <td>522717</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>MT-ND5</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>Conflicting_classifications_of_pathogenicity</td>\n",
       "      <td>criteria_provided,_conflicting_classifications</td>\n",
       "      <td>Likely_pathogenic(1)|Benign(1)</td>\n",
       "      <td>Mitochondrial_disease|not_specified</td>\n",
       "      <td>MONDO:MONDO:0044970,MeSH:D028361,MedGen:C07516...</td>\n",
       "      <td>COI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2725769</th>\n",
       "      <td>MT</td>\n",
       "      <td>14597</td>\n",
       "      <td>209173</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>MT-ND6</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>Conflicting_classifications_of_pathogenicity</td>\n",
       "      <td>criteria_provided,_conflicting_classifications</td>\n",
       "      <td>Likely_pathogenic(1)|Uncertain_significance(2)</td>\n",
       "      <td>not_specified|Dystonic_disorder|Dysarthria|Lei...</td>\n",
       "      <td>MedGen:CN169374|Human_Phenotype_Ontology:HP:00...</td>\n",
       "      <td>COI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2725770</th>\n",
       "      <td>MT</td>\n",
       "      <td>14598</td>\n",
       "      <td>374080</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>MT-ND6</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>Conflicting_classifications_of_pathogenicity</td>\n",
       "      <td>criteria_provided,_conflicting_classifications</td>\n",
       "      <td>Likely_pathogenic(1)|Likely_benign(1)</td>\n",
       "      <td>Parkinsonian_disorder|Blindness|Leigh_syndrome</td>\n",
       "      <td>Human_Phenotype_Ontology:HP:0001300,MONDO:MOND...</td>\n",
       "      <td>COI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2726151</th>\n",
       "      <td>MT</td>\n",
       "      <td>15923</td>\n",
       "      <td>39575</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>PASS</td>\n",
       "      <td>MT-TT</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>Conflicting_classifications_of_pathogenicity</td>\n",
       "      <td>criteria_provided,_conflicting_classifications</td>\n",
       "      <td>Pathogenic(1)|Uncertain_significance(1)</td>\n",
       "      <td>Mitochondrial_disease|Variant_of_unknown_signi...</td>\n",
       "      <td>MONDO:MONDO:0044970,MeSH:D028361,MedGen:C07516...</td>\n",
       "      <td>COI</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>120452 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        CHROM     POS       ID REF ALT  QUAL FILTER GENEINFO  \\\n",
       "37          1  930200  1043045   G   A -10.0   PASS   SAMD11   \n",
       "151         1  935839  1085785   C   T -10.0   PASS   SAMD11   \n",
       "214         1  939117  1427749   G   A -10.0   PASS   SAMD11   \n",
       "325         1  941284  1006858   G   A -10.0   PASS   SAMD11   \n",
       "438         1  942470  2334732   G   A -10.0   PASS   SAMD11   \n",
       "...       ...     ...      ...  ..  ..   ...    ...      ...   \n",
       "2725349    MT   12278   690182   T   C -10.0   PASS   MT-TL2   \n",
       "2725384    MT   12372   522717   G   A -10.0   PASS   MT-ND5   \n",
       "2725769    MT   14597   209173   A   G -10.0   PASS   MT-ND6   \n",
       "2725770    MT   14598   374080   T   C -10.0   PASS   MT-ND6   \n",
       "2726151    MT   15923    39575   A   G -10.0   PASS    MT-TT   \n",
       "\n",
       "                             CLNVC  \\\n",
       "37       single_nucleotide_variant   \n",
       "151      single_nucleotide_variant   \n",
       "214      single_nucleotide_variant   \n",
       "325      single_nucleotide_variant   \n",
       "438      single_nucleotide_variant   \n",
       "...                            ...   \n",
       "2725349  single_nucleotide_variant   \n",
       "2725384  single_nucleotide_variant   \n",
       "2725769  single_nucleotide_variant   \n",
       "2725770  single_nucleotide_variant   \n",
       "2726151  single_nucleotide_variant   \n",
       "\n",
       "                                               CLNSIG  \\\n",
       "37       Conflicting_classifications_of_pathogenicity   \n",
       "151      Conflicting_classifications_of_pathogenicity   \n",
       "214      Conflicting_classifications_of_pathogenicity   \n",
       "325      Conflicting_classifications_of_pathogenicity   \n",
       "438      Conflicting_classifications_of_pathogenicity   \n",
       "...                                               ...   \n",
       "2725349  Conflicting_classifications_of_pathogenicity   \n",
       "2725384  Conflicting_classifications_of_pathogenicity   \n",
       "2725769  Conflicting_classifications_of_pathogenicity   \n",
       "2725770  Conflicting_classifications_of_pathogenicity   \n",
       "2726151  Conflicting_classifications_of_pathogenicity   \n",
       "\n",
       "                                             CLNREVSTAT  \\\n",
       "37       criteria_provided,_conflicting_classifications   \n",
       "151      criteria_provided,_conflicting_classifications   \n",
       "214      criteria_provided,_conflicting_classifications   \n",
       "325      criteria_provided,_conflicting_classifications   \n",
       "438      criteria_provided,_conflicting_classifications   \n",
       "...                                                 ...   \n",
       "2725349  criteria_provided,_conflicting_classifications   \n",
       "2725384  criteria_provided,_conflicting_classifications   \n",
       "2725769  criteria_provided,_conflicting_classifications   \n",
       "2725770  criteria_provided,_conflicting_classifications   \n",
       "2726151  criteria_provided,_conflicting_classifications   \n",
       "\n",
       "                                             CLNSIGCONF  \\\n",
       "37           Uncertain_significance(1)|Likely_benign(1)   \n",
       "151          Uncertain_significance(1)|Likely_benign(1)   \n",
       "214          Uncertain_significance(1)|Likely_benign(1)   \n",
       "325          Uncertain_significance(1)|Likely_benign(1)   \n",
       "438          Uncertain_significance(1)|Likely_benign(1)   \n",
       "...                                                 ...   \n",
       "2725349  Likely_pathogenic(1)|Uncertain_significance(2)   \n",
       "2725384                  Likely_pathogenic(1)|Benign(1)   \n",
       "2725769  Likely_pathogenic(1)|Uncertain_significance(2)   \n",
       "2725770           Likely_pathogenic(1)|Likely_benign(1)   \n",
       "2726151         Pathogenic(1)|Uncertain_significance(1)   \n",
       "\n",
       "                                                     CLNDN  \\\n",
       "37                    not_provided|Inborn_genetic_diseases   \n",
       "151                   Inborn_genetic_diseases|not_provided   \n",
       "214                   not_provided|Inborn_genetic_diseases   \n",
       "325                             not_specified|not_provided   \n",
       "438                   not_provided|Inborn_genetic_diseases   \n",
       "...                                                    ...   \n",
       "2725349  Juvenile_myopathy,_encephalopathy,_lactic_acid...   \n",
       "2725384                Mitochondrial_disease|not_specified   \n",
       "2725769  not_specified|Dystonic_disorder|Dysarthria|Lei...   \n",
       "2725770     Parkinsonian_disorder|Blindness|Leigh_syndrome   \n",
       "2726151  Mitochondrial_disease|Variant_of_unknown_signi...   \n",
       "\n",
       "                                                  CLNDISDB CLNSIG_MASK  \n",
       "37            MedGen:CN517202|MeSH:D030342,MedGen:C0950123         COI  \n",
       "151           MeSH:D030342,MedGen:C0950123|MedGen:C3661900         COI  \n",
       "214           MedGen:C3661900|MeSH:D030342,MedGen:C0950123         COI  \n",
       "325                        MedGen:CN169374|MedGen:C3661900         COI  \n",
       "438           MedGen:C3661900|MeSH:D030342,MedGen:C0950123         COI  \n",
       "...                                                    ...         ...  \n",
       "2725349  MONDO:MONDO:0010789,MedGen:C0162671,OMIM:54000...         COI  \n",
       "2725384  MONDO:MONDO:0044970,MeSH:D028361,MedGen:C07516...         COI  \n",
       "2725769  MedGen:CN169374|Human_Phenotype_Ontology:HP:00...         COI  \n",
       "2725770  Human_Phenotype_Ontology:HP:0001300,MONDO:MOND...         COI  \n",
       "2726151  MONDO:MONDO:0044970,MeSH:D028361,MedGen:C07516...         COI  \n",
       "\n",
       "[120452 rows x 15 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cv_dict['clinvar_20240407'][cv_dict['clinvar_20240407']['CLNSIG_MASK']=='COI']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "258fdc46",
   "metadata": {},
   "source": [
    "# Filter genes\n",
    "We further restricted the data set to those genes annotated as having robust association with rare disease reported in OMIM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "62bd07de",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene stable ID</th>\n",
       "      <th>Gene name</th>\n",
       "      <th>MIM morbid accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>MT-TF</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000210077</td>\n",
       "      <td>MT-TV</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene stable ID Gene name MIM morbid accession\n",
       "0  ENSG00000210049     MT-TF                     \n",
       "1  ENSG00000211459   MT-RNR1                     \n",
       "2  ENSG00000210077     MT-TV                     "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# BioMart data\n",
    "gene_mim = pd.read_csv(f'{path_to_data}/features/bioMart_Gene_MIM.txt', sep='\\t', keep_default_na=False)\n",
    "gene_mim.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "93477ea3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4786"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_mim = gene_mim[gene_mim['MIM morbid accession']!= '']\n",
    "gene_mim_glist = list(gene_mim['Gene name'].drop_duplicates())\n",
    "len(gene_mim_glist)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "764f9cf0",
   "metadata": {},
   "outputs": [],
   "source": [
    "for vcf in cv_dict:\n",
    "    cv_dict[vcf] = cv_dict[vcf][cv_dict[vcf]['GENEINFO'].isin(gene_mim_glist)]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b533f8ad",
   "metadata": {},
   "source": [
    "# Collect Metrics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "fa5ed773",
   "metadata": {},
   "outputs": [],
   "source": [
    "def collect_ids_and_metrics(data_dict, metric):\n",
    "    \n",
    "    \"\"\"\n",
    "    Collects metrics for each VariantID across dfs collected in the dictionary\n",
    "    \n",
    " Args: \n",
    "    data_dict: A dictionary, which keys (sorted) are names of ClinVar vcfs \n",
    "    and items are pandas dfs with nessesary columns for the analysis \n",
    "    metric: A column (CLNSIG or CLNREVSTAT) which should be summaerized per each VariantID across all used vcfs\n",
    "    \n",
    "  Returns:\n",
    "     A dictionary where keys are variant IDs. Values are all CLNSIG or CLNREVSTAT records\n",
    "    \n",
    "    \"\"\"\n",
    "    \n",
    "    all_ids = set()\n",
    "    metrics_dict = {}\n",
    "    \n",
    "    for df in data_dict.values():\n",
    "         all_ids.update(set(df['ID'].unique()))\n",
    "\n",
    "        \n",
    "    for year, df in data_dict.items():\n",
    "        print(year)\n",
    "        for id in all_ids:\n",
    "            if id in df['ID'].to_numpy():\n",
    "                metric_value = df[df['ID'] == id][f'{metric}'].tolist()[0]\n",
    "                metrics_dict[id] = metrics_dict.get(id, []) + [metric_value]\n",
    "            else:\n",
    "                metrics_dict[id] = metrics_dict.get(id, []) + ['NA']\n",
    "    \n",
    "    return metrics_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "9b6b8ccc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "clinvar_20180401\n",
      "clinvar_20180930\n",
      "clinvar_20190403\n",
      "clinvar_20191001\n",
      "clinvar_20200407\n",
      "clinvar_20200927\n",
      "clinvar_20210328\n",
      "clinvar_20210929\n",
      "clinvar_20220403\n",
      "clinvar_20221001\n",
      "clinvar_20230404\n",
      "clinvar_20230930\n",
      "clinvar_20240407\n"
     ]
    }
   ],
   "source": [
    "id_clnsig = collect_ids_and_metrics(cv_dict, 'CLNSIG_MASK')\n",
    "\n",
    "with open(f'{path_to_temp}/id_clnsig.pkl', 'wb') as file:   # save CLNSIG dict\n",
    "    pickle.dump(id_clnsig, file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c2942dd3",
   "metadata": {},
   "outputs": [],
   "source": [
    "#id_revstat = collect_ids_and_metrics(cv_dict, 'CLNREVSTAT')\n",
    "\n",
    "#with open(f'{path_to_temp}/id_revstat.pkl', 'wb') as file:   # save CLNREVSTAT dict\n",
    "#    pickle.dump(id_revstat, file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "b5b389ab",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "clinvar_20180401\n",
      "clinvar_20180930\n",
      "clinvar_20190403\n",
      "clinvar_20191001\n",
      "clinvar_20200407\n",
      "clinvar_20200927\n",
      "clinvar_20210328\n",
      "clinvar_20210929\n",
      "clinvar_20220403\n",
      "clinvar_20221001\n",
      "clinvar_20230404\n",
      "clinvar_20230930\n",
      "clinvar_20240407\n"
     ]
    }
   ],
   "source": [
    "id_clnsigconf = collect_ids_and_metrics(cv_dict, 'CLNSIGCONF')\n",
    "\n",
    "with open(f'{path_to_temp}/id_clnsigconf.pkl', 'wb') as file:   # save CLNSIGCONF dict\n",
    "    pickle.dump(id_clnsigconf, file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "e6b95c91",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(f'{path_to_temp}/id_clnsig.pkl', 'rb') as file:   # load CLNSIG dict\n",
    "    id_clnsig = pickle.load(file)\n",
    "\n",
    "#with open(f'{path_to_temp}/id_revstat.pkl', 'rb') as file:   # load CLNREVSTAT dict\n",
    "    #id_revstat = pickle.load(file)\n",
    "    \n",
    "with open(f'{path_to_temp}/id_clnsigconf.pkl', 'rb') as file:   # load CLNREVSTAT dict\n",
    "    id_clnsigconf1 = pickle.load(file)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "feb2936f",
   "metadata": {},
   "source": [
    "# Collect COI * variants\n",
    "\n",
    "*genetic variants with conflicting interpretations of pathogenicity"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "a07984a9",
   "metadata": {},
   "outputs": [],
   "source": [
    "def filter_conflicting_records(clnsig_data, coi_list):\n",
    "  \n",
    "  \"\"\"\n",
    "  Filters a ClinVar data dictionary to include only records with conflicting interpretations.\n",
    "\n",
    "  Args:\n",
    "      clnsig_data: A dictionary where keys are variant IDs and values are sorted lists of clinical significance records.\n",
    "      conflicts_clnsig_records: A list of strings representing conflicting interpretation records.\n",
    "\n",
    "  Returns:\n",
    "      A dictionary containing only variants with at least one record from conflicts_clnsig_records. \n",
    "      The value for each variant remains the original list of clinical significance records.\n",
    "  \"\"\"\n",
    "  \n",
    "  flt_dict = {}\n",
    "  for variant_id, clnsig_records in clnsig_data.items():\n",
    "     for record in clnsig_records:\n",
    "        if record in coi_list:\n",
    "            flt_dict[variant_id] = clnsig_records  # Include entire record list\n",
    "            break  # Stop iterating after finding a conflicting record\n",
    "  \n",
    "  return flt_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "3792e4bd",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Variants of records representing emerged conflict\n",
    "\n",
    "list_conflict_records = ['Conflicting_interpretations_of_pathogenicity',\n",
    "'Conflicting_interpretations_of_pathogenicity,_Affects',\n",
    "'Conflicting_interpretations_of_pathogenicity,_Affects,_association,_drug_response,_other',\n",
    "'Conflicting_interpretations_of_pathogenicity,_Affects,_association,_risk_factor',\n",
    "'Conflicting_interpretations_of_pathogenicity,_Affects,_other',\n",
    "'Conflicting_interpretations_of_pathogenicity,_association',\n",
    "'Conflicting_interpretations_of_pathogenicity,_association,_other,_risk_factor',\n",
    "'Conflicting_interpretations_of_pathogenicity,_association,_risk_factor',\n",
    "'Conflicting_interpretations_of_pathogenicity,_drug_response',\n",
    "'Conflicting_interpretations_of_pathogenicity,_drug_response,_other',\n",
    "'Conflicting_interpretations_of_pathogenicity,_other',\n",
    "'Conflicting_interpretations_of_pathogenicity,_other,_risk_factor',\n",
    "'Conflicting_interpretations_of_pathogenicity,_protective',\n",
    "'Conflicting_interpretations_of_pathogenicity,_risk_factor',\n",
    "'Conflicting_interpretations_of_pathogenicity|Affects',\n",
    "'Conflicting_interpretations_of_pathogenicity|_Affects',\n",
    "'Conflicting_interpretations_of_pathogenicity|_association',\n",
    "'Conflicting_interpretations_of_pathogenicity|_association|_risk_factor',\n",
    "'Conflicting_interpretations_of_pathogenicity|_drug_response',\n",
    "'Conflicting_interpretations_of_pathogenicity|_drug_response|_other',\n",
    "'Conflicting_interpretations_of_pathogenicity|_other',\n",
    "'Conflicting_interpretations_of_pathogenicity|_other|_risk_factor',\n",
    "'Conflicting_interpretations_of_pathogenicity|_protective',\n",
    "'Conflicting_interpretations_of_pathogenicity|_risk_factor',\n",
    "'Conflicting_interpretations_of_pathogenicity|association',\n",
    "'Conflicting_interpretations_of_pathogenicity|association|other',\n",
    "'Conflicting_interpretations_of_pathogenicity|association|risk_factor',\n",
    "'Conflicting_interpretations_of_pathogenicity|drug_response',\n",
    "'Conflicting_interpretations_of_pathogenicity|drug_response|other',\n",
    "'Conflicting_interpretations_of_pathogenicity|other',\n",
    "'Conflicting_interpretations_of_pathogenicity|other|risk_factor',\n",
    "'Conflicting_interpretations_of_pathogenicity|protective',\n",
    "'Conflicting_interpretations_of_pathogenicity|risk_factor', \n",
    "'Conflicting_classifications_of_pathogenicity',\n",
    "'Conflicting_classifications_of_pathogenicity|Affects',\n",
    "'Conflicting_classifications_of_pathogenicity|association',\n",
    "'Conflicting_classifications_of_pathogenicity|association|risk_factor',\n",
    "'Conflicting_classifications_of_pathogenicity|drug_response',\n",
    "'Conflicting_classifications_of_pathogenicity|drug_response|other',\n",
    "'Conflicting_classifications_of_pathogenicity|other',\n",
    "'Conflicting_classifications_of_pathogenicity|other|risk_factor',\n",
    "'Conflicting_classifications_of_pathogenicity|protective',\n",
    "'Conflicting_classifications_of_pathogenicity|risk_factor',]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "77afb5b5",
   "metadata": {},
   "outputs": [],
   "source": [
    "clnsigmask_coi = ['COI']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "7ab3860e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All variants:2320593\n",
      "Conflicting interpreatation: 131319\n"
     ]
    }
   ],
   "source": [
    "coi_clnsig = filter_conflicting_records(id_clnsig, clnsigmask_coi)\n",
    "\n",
    "with open(f'{path_to_temp}/coi_clnsig.pkl', 'wb') as file:   # save   dict\n",
    "    pickle.dump(coi_clnsig, file)\n",
    "    \n",
    "print(f'All variants:{len(id_clnsig)}')\n",
    "print(f'Conflicting interpreatation: {len(coi_clnsig)}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "73d0696e",
   "metadata": {},
   "outputs": [],
   "source": [
    "#  Make df from dictionary\n",
    "coi_clnsig_df = pd.DataFrame.from_dict(coi_clnsig, orient='columns')\n",
    "coi_clnsig_df = coi_clnsig_df.transpose().reset_index(names = ['varid'])\n",
    "coi_clnsig_df.columns = [\"varid\"] + vcfs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "3d33a003",
   "metadata": {},
   "outputs": [],
   "source": [
    "# List of COI IDs to filter by\n",
    "coi_ids_list = coi_clnsig.keys()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
